modin/core/storage_formats/pandas/parsers.py

```diff
@@ -585,7 +585,11 @@ def update_row_nums(match):
             **kwargs,
         )
         pandas_df = parser.read()
-        if len(pandas_df) > 1 and pandas_df.isnull().all().all():
+        if (
+            len(pandas_df) > 1
+            and len(pandas_df.columns) != 0
```

Comment:
what is the point of this check? did this blow up if an empty df but with some index is present, or what?

also do we have a test for this?

Reply:
Yes, deleting the index causes the test to fail. The one for which I removed `@pytest.mark.xfail`.

Reply:
okay, I got it, this is intended to support some kind of a weird use case where all we want to read from an excel is just an index:
```python
>>> pandas.read_excel("test.xlsx", index_col=0, usecols=[0])
Empty DataFrame
Columns: []
Index: [0, 1, 2, 3]
```
such kind of a frame would have hit the previous version of this if-condition and would become something like:
```python
>>> pandas.DataFrame(columns=df.columns)
Empty DataFrame
Columns: []
Index: []
```